lib/progress_bar/enumerable.rb

```diff
@@ -0,0 +1,6 @@
+require_relative '../progress_bar'
+
+# FIXME: should there be a better method?..
+[Enumerable, Array, Hash, Range].each do |mod|
```

Comment:
If we just directly monkey-patch `Enumerable`, then `#with_progress` will also appear in all classes where its included.
But unfortunately, if we just include something in `Enumerable`, it will not be included recursively in `Array`, `Hash` and so on. So, it seems there's no "clean" solution -- either monkey-patching (which is typically considered bad practice), or  repetitive inclusion of `WithProgress`.